### ISSUE TITLE
Editor Modes: Navigation in Select Mode doesn't work smoothly with nested blocks

### ISSUE THREAD

**Issue by gziolo** [WordPress/gutenberg#18780]
**Describe the bug**
Upon entering "Navigation" mode from the nested block, you got stuck in the context of the parent block. You can't use arrow up/down keys to navigate to the parent block. In addition, if there is only one child block, then those keys don't work at all and sometimes the focus gets lost. In effect, you can't enter "Edit" mode anymore with the keyboard.

The same issues exist when tabbing after "Navigation" mode gets enabled for the nested block. The behavior is slightly different. Pressing `Tab` moves the focus to the Publish Panel. Pressing `Shift` + `Tab` moves the focus to the block controls which shouldn't show up in this mode.

**Expected behavior**
Both arrow up/down keys and tabbing works properly and respects both children and parent blocks.

**Screenshots**
![keyboard-modes](https://user-images.githubusercontent.com/699132/69726465-efeaf580-1120-11ea-9acd-ff401d0c0ebe.gif)
![keyboard-modes-tab](https://user-images.githubusercontent.com/699132/69726482-f8433080-1120-11ea-9707-0de93abf7614.gif)

**Desktop (please complete the following information):**
 - OS: macOS
 - Browser Safari
 - Version 13

**Additional context**
Gutenberg 7.0 - the latest master.

**Comment by gziolo** [comment]
I tested again, and it seems to work as expected now. Arrow keys navigation is locked down to it's immediate parent, but tabbing allows to go to the pre/next block regardless of the context.
